tests/mock/Concrete/PlatformOrderDecorator.php

```diff
@@ -0,0 +1,899 @@
+
+        return $this->$method($quote);
+    }
+
+
+    private function getRegisteredCustomer($quote)
+    {
+        $quoteCustomer = $quote->getCustomer();
+
+        $addresses = $quoteCustomer->getAddresses();
+        $address = end($addresses);
+
+        $customerRepository =
+            ObjectManager::getInstance()->get(CustomerRepository::class);
+        $savedCustomer = $customerRepository->getById($quoteCustomer->getId());
+
+        $customer = new Customer;
+        $customer->setCode($savedCustomer->getId());
+
+        $mpId = null;
+        try {
+            $mpId = $savedCustomer->getCustomAttribute('customer_id_mundipagg')
+                ->getValue();
+            $customerId = new CustomerId($mpId);
+            $customer->setMundipaggId($customerId);
+        } catch (\Throwable $e) {
+        }
+
+        if (empty($mpId)) {
+            $coreCustomerRespository = new CoreCustomerRepository();
+            $coreCustomer = $coreCustomerRespository->findByCode(
+                $savedCustomer->getId()
+            );
+            if ($coreCustomer !== null) {
+                $customer->setMundipaggId($coreCustomer->getMundipaggId());
+            }
+        }
+
+        $fullName = implode(' ', [
+            $quote->getCustomerFirstname(),
+            $quote->getCustomerMiddlename(),
+            $quote->getCustomerLastname(),
+        ]);
+
+        $fullName = preg_replace("/  /", " ", $fullName);
+
+        $customer->setName($fullName);
+        $customer->setEmail($quote->getCustomerEmail());
+
+        $cleanDocument = preg_replace(
+            '/\D/',
+            '',
+            $quote->getCustomerTaxvat()
+        );
+
+        if (empty($cleanDocument)) {
+            $cleanDocument = preg_replace(
+                '/\D/',
+                '',
+                $address->getVatId()
+            );
+        }
+
+        $customer->setDocument($cleanDocument);
+        $customer->setType(CustomerType::individual());
+
+        $telephone = $address->getTelephone();
+        $phone = new Phone($telephone);
+
+        $customer->setPhones(
+            CustomerPhones::create([$phone, $phone])
+        );
+
+        $address = $this->getAddress($address);
+
+        $customer->setAddress($address);
+
+        return $customer;
+
+    }
+
+    private function getGuestCustomer($quote)
+    {
+        $guestAddress = $quote->getBillingAddress();
+
+        $customer = new Customer;
+
+        $customer->setName($guestAddress->getName());
+        $customer->setEmail($guestAddress->getEmail());
+
+        $cleanDocument = preg_replace(
+            '/\D/',
+            '',
+            $guestAddress->getVatId()
+        );
+
+        $customer->setDocument($cleanDocument);
+        $customer->setType(CustomerType::individual());
+
+        $telephone = $guestAddress->getTelephone();
+        $phone = new Phone($telephone);
+
+        $customer->setPhones(
+            CustomerPhones::create([$phone, $phone])
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class '\Mundipagg\Core\Payment\ValueObjects\CustomerPhones' in method 'getGuestCustomer'.](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=4926694)